API/utils/Model.php

```diff
@@ -811,6 +811,121 @@ function RemoveAdmin($email) {
+				VALUES (?, ?, ?, ?);";
+            $stmt = $this->PrepareAndExecute($sql, array($term, $projectId, $textBackTranslation, $generatedFileName));
+            $this->FreeStmt($stmt);
+
+            $recordingId = $this->conn->lastInsertId();
+            $this->conn->commit();
+
+            # save recording file content
+            $directory = "Projects/$androidId/WordLinks";
+            PutFile($directory, $generatedFileName, $audioData);
+
+            return $recordingId;
+        } catch (\Exception $e) {
+            error_log($e->getMessage());
+            $this->conn->rollBack();
+            RespondWithError(500, $e->getMessage());
+        }
+    }
+
+    /**
+     * @param int $recordingId
+     * @param int $projectId
+     * @param string $androidId
+     * @param string $term
+     * @param string $textBackTranslation
+     * @param string $audioRecordingFilename
+     * @param string $audioData
+     * @return int
+     */
+    public function UpdateWordLinkRecording($recordingId, $projectId, $androidId, $term, $textBackTranslation, $audioRecordingFilename, $audioData)
+    {
+         try {
+             # check a recording exists for given recordingId, term & projectId
+             $sql = "SELECT id, fileName FROM WordLinkRecordings WHERE id = ? AND term = ? AND projectId = ?;";
+             $stmt = $this->PrepareAndExecute($sql, array($recordingId, $term, $projectId));
+
+             $recording = $this->FetchArray($stmt);
+             $this->FreeStmt($stmt);
+
+             if (!is_array($recording)) {
+                 RespondWithError(400, "No WordLinkRecordings found for given recordingID: '$recordingId', term: '$term' and androidId: '$androidId'");
+             }
+
+             $existingRecodingFileName = $recording['fileName'];
+
+             # generate random file for saving uploaded recording file content
+             $generatedFileName = $this->GenerateRecordingFileName($audioRecordingFilename);
+
+             # update WordLinkRecordings
+             $this->conn->beginTransaction();
+             $updateSql =  "UPDATE WordLinkRecordings
+                            SET textBackTranslation = ?, fileName = ? WHERE id = ?";
+
+             $stmt = $this->PrepareAndExecute($updateSql, array($textBackTranslation, $generatedFileName, $recordingId));
+             $this->FreeStmt($stmt);
+
+             $this->conn->commit();
+             $directory = "Projects/$androidId/WordLinks";
+             PutFile($directory, $generatedFileName, $audioData);
+
+             # delete existing recording file
+             $existingRecodingFilepath = "{$GLOBALS['filesRoot']}/$directory/$existingRecodingFileName";
+
+             error_log(sprintf("Deleting old recording file '%s'", $existingRecodingFilepath));
+             if(!unlink($existingRecodingFilepath)) {
+                 throw new \Exception("Failed to deleted existing recording file: $existingRecodingFilepath");
```

Comment:
Let's log the error but don't throw. It's undesirable to not delete the file, but it's acceptable to have orphaned files.